cylc/uiserver/jupyterhub_config.py

```diff
@@ -49,21 +58,40 @@ def _load(path):
         exec(path.read_text())
 
 
-def load():
+def get_conf_dir_hierarchy(
+        config_paths: List[Path], filename: bool = True
+) -> List[Path]:
+    """Takes list of config paths, adds version and filename to the path"""
+    conf_hierarchy = []
+    version_hierarchy = get_version_hierarchy(hub_version or __version__)
+    for x in product(config_paths, version_hierarchy):
+        if filename:
+            conf_hierarchy.append(Path(x[0], x[1], 'jupyter_config.py'))
+        else:
+            conf_hierarchy.append(Path(x[0], x[1]))
+    return conf_hierarchy
```

Comment:
Can avoid the `map(str(` in `cylc.uiserver.app` by:

```suggestion
    return str(conf_hierarchy)
```